### PR TITLE
Get-DecryptedObject - Fix some issues

### PR DESCRIPTION
**Pull request by andreasjordan**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes # )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/dataplat/appveyor-lab ?
 - [ ] Unit test is included
 - [ ] Documentation
 - [ ] Build system

Found some issues. Not sure if related to the problem recently reported on Slack - but maybe.

Main issue was that the catch block related to the DAC was outside of the Invoke-Command2 - moved it to the inside. Also fixed some minor issies with other try-catch blocks and wording of messaged.

### CODE REVIEW

**Comment by andreasjordan** [comment]
Now this works:
```
$dac = Connect-DbaInstance -SqlInstance sql01 -DedicatedAdminConnection
Copy-DbaCredential -Source $dac -Destination sql02
```
This way, we don't need to open a DAC via Invoke-Command2. But works only, if remote DAC is enabled on sql01.

File: functions/Copy-DbaCredential.ps1 [hunk 1]
```diff
@@ -131,7 +131,13 @@ function Copy-DbaCredential {
             )
 
             Write-Message -Level Verbose -Message "Collecting Credential logins and passwords on $($sourceServer.Name)"
-            $sourceCredentials = Get-DecryptedObject -SqlInstance $sourceServer -Type Credential
+            try {
+                $sourceCredentials = Get-DecryptedObject -SqlInstance $sourceServer -Type Credential -EnableException
+            } catch {
+                Stop-Function -Message "Failed to decrypted credentials" -ErrorRecord $_
```
Comment (potatoqualitee) [inline functions/Copy-DbaCredential.ps1:137]: lookin good so far!

```suggestion
                Stop-Function -Message "Failed to decrypt credentials on $($sourceServer.Name)" -ErrorRecord $_
```

**Comment by potatoqualitee** [comment]
Thanks so much, looking forward to this merge. 

Do you know math? I don't 😅  But if you do, any interest in looking into how to decrypt Database Mail Passwords?

**Comment by potatoqualitee** [comment]
LGTM! and tests pass

**Comment by andreasjordan** [comment]
> Do you know math? I don't 😅 But if you do, any interest in looking into how to decrypt Database Mail Passwords?

I know a bit of math, but I don't use Database Mail and don't have a lab for that. So I have to say no for the moment. But you can put all you have in an issue and maybe I find time later...

**Pull request merged**